### PR TITLE
Softfail httpd container start

### PR DESCRIPTION
**Pull request by mloviska**
In sle-micro 5.1 httpd container fails to start when SELinux is turned on. Introduce a check whether the container started for netavark test are running.

#### VRs: 

* [sle-micro-5.1-MicroOS-Image-Updates-x86_64-Build20230528-1-slem_containers@64bit](http://kepler.suse.cz/tests/20983#step/podman_netavark/1)
* [sle-micro-5.2-MicroOS-Image-Updates-x86_64-Build20230528-1-slem_containers_selinux@64bit](http://kepler.suse.cz/tests/20982#step/podman_netavark/1)
* [sle-micro-5.4-Default-Updates-x86_64-Build20230528-1-slem_containers@64bit](http://kepler.suse.cz/tests/20979#step/podman_netavark/1)
* [sle-micro-5.3-MicroOS-Image-Updates-x86_64-Build20230528-1-slem_containers_selinux@64bit](http://kepler.suse.cz/tests/20981#step/podman_netavark/1)
* [sle-micro-5.1-MicroOS-Image-Updates-x86_64-Build20230528-1-slem_containers_selinux@64bit](http://kepler.suse.cz/tests/20980#step/podman_netavark/132)

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]
LGTM

File: tests/containers/podman_netavark.pm [hunk 1]
```diff
@@ -113,17 +127,19 @@ sub run {
...
+    if (is_container_running($ctr1->{name}, $ctr2->{name})) {
+        my $net1_reg = qr@ether\s+$ctr2->{mac}.*\s+inet\s+$ctr2->{ip}\/16@;
+        my $net2_reg = qr@ether\s+$ctr2->{mac_sec}.*\s+inet\s+$ctr2->{ip_sec}\/16@;
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth0'", sub { /$net1_reg|$net2_reg/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth1'", sub { /$net1_reg|$net2_reg/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'nslookup $ctr1->{name}'", sub { /Name:\s+$ctr1->{name}\s+Address.*$ctr1->{ip}/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'wget -S $ctr1->{ip}:80'", sub { /HTTP.* 200 OK/ });
+        # busybox container should be able to resolve apache container
+        assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+        assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
```
Comment (grisu48) [inline tests/containers/podman_netavark.pm:139]: Is this correct? I would assume this would fail due to the different interface names, no?

**Review by grisu48 — COMMENTED** [review]
Some suggestions, overall LGTM!

File: tests/containers/podman_netavark.pm (hunk 1 above)
Comment (mloviska) [reply tests/containers/podman_netavark.pm:139]: No, both of the containers should be in the same network segment

File: tests/containers/podman_netavark.pm [hunk 2]
```diff
@@ -86,8 +98,10 @@ sub run {
     assert_script_run("podman network create --gateway $net1->{gateway} --subnet $net1->{subnet} $net1->{name}");
     assert_script_run("podman run --network $net1->{name}:ip=$ctr1->{ip},mac=$ctr1->{mac} -d --name $ctr1->{name} $ctr1->{image}");
     assert_script_run("podman container inspect $ctr1->{name} --format {{.NetworkSettings.Networks.$net1->{name}.IPAddress}}");
-    validate_script_output("curl --head --silent $ctr1->{ip}:80", sub { /HTTP.* 200 OK/ });
-    assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+    if (is_container_running($ctr1->{name})) {
+        validate_script_output("curl --head --silent $ctr1->{ip}:80", sub { /HTTP.* 200 OK/ });
```
[the remark this answers is not in the document]
Comment (mloviska) [reply tests/containers/podman_netavark.pm:102]: I think, that was the initial proposal already. :D

File: tests/containers/podman_netavark.pm [hunk 3]
```diff
@@ -23,6 +23,18 @@ sub remove_subtest_setup {
     validate_script_output("podman ps -a --noheading", sub { /^\s*$/ });
 }
 
+sub is_container_running {
+    my @containers = @_;
+    my $out = script_output("podman container ps --format '{{.Names}}' --noheading");
+
+    foreach my $cont (@containers) {
+        next if ($out =~ m/$cont/);
+        record_soft_failure('bsc#1211774 - podman fails to start container with SELinux');
```
[the remark this answers is not in the document]
Comment (mloviska) [reply tests/containers/podman_netavark.pm:32]: I agree, and also the proper check should come only against the `httpd` container. As this happens only on slem 5.1. I would rather to keep it simple, as we will need to remove only the `record_soft_failure` line.

**Review by pdostal — APPROVED** [review]

File: tests/containers/podman_netavark.pm (hunk 1 above)
Comment (grisu48) [reply tests/containers/podman_netavark.pm:139]: Ah ok, thnx!

**Review by grisu48 — APPROVED** [review]

**Pull request merged**